src/profile-logic/profile-data.js

```diff
@@ -2683,3 +2683,14 @@ export function getThreadsKey(threadIndexes: Set<ThreadIndex>): ThreadsKey {
 
   return [...threadIndexes].sort((a, b) => b - a).join(',');
 }
+
+/**
+ * Checks if threadIndexesSet contains all the threads in the threadsKey.
+ */
+export function hasThreadKeys(
```

Comment:
Optional: So this runs on every single render update for the component, so it might be kind of hot. It's not much code, but it runs very very frequently. It might be worth considering a selector, or transform it into imperative code that may run a bit faster. I'm happy to leave it up to your judgement if you just want to leave it as-is.

Reply:
Thanks for chatting on the Matrix channel. Now we are doing the memoization for `hasThreadKeys` usages. Also created #2787 for the `Set` issue we discussed.